docs/app/components/components.component.ts

```diff
@@ -1,15 +1,85 @@
-import { Component, OnInit } from '@angular/core';
+import { Component } from '@angular/core';
+import { Router, NavigationEnd } from '@angular/router';
 
 @Component({
     selector: 'hc-demo-components',
     templateUrl: './components.component.html'
 })
-export class ComponentsComponent implements OnInit {
+export class ComponentsComponent {
 
-    constructor() {
-    }
+    thisPage = 'Accordion';
 
-    ngOnInit() {
+    constructor( private router: Router ) {
+        // Listen for vertical tab bar navigation and update the select component
+        router.events.subscribe((event) => {
+            if (event instanceof NavigationEnd) {
```

Comment:
Am I correct in understanding that this is listening for the current route and setting the current page? I'm wondering if injecting `private route: ActivatedRoute` and subscribing to the url would be simpler than listening to all of the router events.

Reply:
agreed, this is a strange way to do this. Another option is you can add "data" to routes, so their is extra metadata when you navigate to any page https://angular.io/guide/router
![image](https://user-images.githubusercontent.com/1934453/36680815-139d4a72-1ad4-11e8-82a4-d258221fc2d9.png)

Reply:
@corytak - it's actually listening to the current route and then setting the responsive Select component value.  The idea is that if I pick "Breadcrumbs" in the vertical tabs, and then I shrink my page down - the responsive subheader with the select component navigation should say "Breadcrumbs" not "Accordion".  I like the idea of adding data to the route to handle this.  I'll probably reach out to either you or Ben to double check my approach before I build this in.